hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/UpsertPartitioner.java

```diff
@@ -140,11 +154,15 @@ private void assignInserts(WorkloadProfile profile, HoodieEngineContext context)
     Map<String, List<SmallFile>> partitionSmallFilesMap =
         getSmallFilesForPartitions(new ArrayList<String>(partitionPaths), context);
 
+    Map<String, Set<String>> partitionPathToPendingClusteringFileGroupsId = getPartitionPathToPendingClusteringFileGroupsId();
+
     for (String partitionPath : partitionPaths) {
       WorkloadStat pStat = profile.getWorkloadStat(partitionPath);
       if (pStat.getNumInserts() > 0) {
+        // exclude the small file in pending clustering, because in pending clustering file not support update now.
+        Set<String> inPendingClusteringFileId = partitionPathToPendingClusteringFileGroupsId.getOrDefault(partitionPath, Collections.emptySet());
+        List<SmallFile> smallFiles = UpdateStrategy.filterSmallFiles(inPendingClusteringFileId, partitionSmallFilesMap.get(partitionPath));
```

Comment:
why is this a static method? If clustering is not enabled on a table, this filtering is expensive. Should we add 'default' strategy for that case? The default strategy would not do any filtering. 

If clustering is enabled, we would use RejectUpdateStrategy and that does filtering.

Reply:
thanks 
1、"why is this a static method?":   because transform updatestrategy into UpsertPartitioner  may be modify the UpsertPartitioner construct.  I will add updatestrategy param to UpsertPartitioner construct
2、" If clustering is not enabled on a table, this filtering is expensive. Should we add 'default' strategy for that case? The default strategy would not do any filtering."  make sense